### PR TITLE
Adds feature related to issue #3

### PR DESCRIPTION
**Pull request by paarandika**
Adds ability to share entries with unique URLs for entries. Not the best way to implement. But I think under the circumstances, this will do.

### CODE REVIEW

**Comment by kasunben** [comment]
Some features related to #3

**Comment by gayanhewa** [comment]
@paarandika rebase to resolve the conflicts.

**Comment by paarandika** [comment]
@gayanhewa To me, it shows that there are no conflicts.

File: app/Http/Controllers/EntryController.php [hunk 1]
```diff
@@ -2,26 +2,38 @@
...
+         $this->need = $needsRepository;
     }
 
     /**
      * List all donatiaons
      *
      * @return $this
      */
-    public function view()
+    public function view($type, $id)
```
Comment (chamilad) [inline app/Http/Controllers/EntryController.php:28]: I don't know PHP that much, Laravel even less. Just out of curiosity, doesn't this need a corresponding method call change that's missing from this PR? Or is that handled through an implicit call somewhere?

**Comment by isuru4lk** [comment]
@paarandika I think we don't need to user a separate controller for entries. We can use the "show' function in both donation and needs controllers and add a condition to check whether it's a ajax request or not.


ex : 
```
public function show(Request $request)
{
   if ($request->ajax()) {
      return "AJAX FOR MODAL";
   }
    return "HTTP FOR ENTRY";
}
```

**Comment by kasunben** [comment]
Let's take this.

**Pull request merged**